### ISSUE TITLE
SNMP table not a table

### ISSUE THREAD

**Issue by phemmer** [influxdata/telegraf#1371]
### Proposal:
There are 2 changes I think would make the SNMP table output more intuitive.

### Current behavior:
Currently the plugin provides output such as:

```
ifHCOutOctets,host=127.0.0.1,instance=enp5s0,unit=octets ifHCOutOctets=10565628i 1456878706044462901
ifInDiscards,host=127.0.0.1,instance=enp5s0 ifInDiscards=0i 1456878706044510264
ifHCInOctets,host=127.0.0.1,instance=enp5s0,unit=octets ifHCInOctets=76351777i 1456878706044531312
```

This is not a table. A table is a key with multiple fields/values.

### Desired behavior:
For example, this is how the `cpu` plugin outputs its data:
```
> cpu,cpu=cpu0,host=fll2azbxstg usage_guest=0,usage_guest_nice=0,usage_idle=95.91836744003224,usage_iowait=0,usage_irq=0,usage_nice=0,usage_softirq=0,usage_steal=0,usage_system=0,usage_user=4.081632652030893 1465880423345219711
> cpu,cpu=cpu1,host=fll2azbxstg usage_guest=0,usage_guest_nice=0,usage_idle=98.0392157292421,usage_iowait=0,usage_irq=0,usage_nice=0,usage_softirq=0,usage_steal=0,usage_system=1.9607843149843065,usage_user=0 1465880423345219711
```
Note how it has a key of `cpu`, with multiple fields. Not a separate line & key for each field.

Thus our earlier table might instead look something like:
```
ifStats,host=127.0.0.1,instance=enp5s0 ifHCOutOctets=10565628i,ifInDiscards=0i,ifHCInOctets=76351777i 1456878706044462901
```

&nbsp;
Now another deviation is the units. In other plugins, such as the disk plugin, the unit is not a tag, it's in the field name. For example:

```
> diskio,host=fll2azbxstg,name=sda2,serial=unknown io_time=23027732i,read_bytes=95604750336i,read_time=31074355i,reads=2188506i,write_bytes=322633367040i,write_time=115839572i,writes=15424778i 1465880806245466582
```

This has several very significant benefits over putting the unit in a tag. First, it lets you use different units for different fields. Second, it lets you put multiple fields on a single line. Third, when graphing, the unit shows up in the graph legend (as a tag, there's no way to get it at all when graphing).

Granted, there's nothing stopping people from putting the unit in the field name now, but providing a `unit` option, which is added a tag is highly confusing. I think it would be much better if it were removed as an explicit option, and instead we provide the option to add arbitrary tags, so if people really want to add it as a tag, they can.

### Use case:
The main thing these changes bring is consistency & expected behavior.
I expect SNMP tables to actually generate a table. This is how it works in every other tool I've worked with which handles SNMP tables. I've never had one generate a separate table for every metric, with only a single field.
It also makes the behavior consistent with other plugins. It's rather odd that this one plugin behaves different from all the other built-in plugins (at least the ones I've used).

**Comment by kostasb** [comment]
Related to #808 .

**Comment by phemmer** [comment]
Closed by #1389
